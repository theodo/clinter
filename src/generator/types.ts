import { Linter } from "eslint";
import { ProjectInfoObject } from "types";

export type ESLintGenerator = (userAnswers: ProjectInfoObject) => (config: Linter.Config) => Linter.Config;

export type ESLintDependencyGenerator = (userAnswers: ProjectInfoObject) => (deps: string[]) => string[];

export interface ESLintOverrider {
  files: string[];
}
