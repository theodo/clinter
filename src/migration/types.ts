export interface MigrationResults {
  fixableErrorCount: number;
  errorCount: number;
}
