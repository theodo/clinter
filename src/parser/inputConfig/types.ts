import { AnswerObject, ModeAnswerObject } from "types";

export interface InputConfig {
  generatorConfig: AnswerObject;
  modeConfig: ModeAnswerObject;
}
