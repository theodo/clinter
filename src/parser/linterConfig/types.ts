import { Options } from "prettier";
import { Linter } from "eslint";

export enum FileExtension {
  JSON = "json",
  YAML = "yml",
  JS = "js",
  NONE = "none",
}

export interface ConfigFileObject {
  name: string;
  extension: FileExtension;
  attribute?: string;
}

export type LinterConfigs = Options | Linter.Config;

export interface ConfigContainer<T extends LinterConfigs> {
  config: T;
  file: ConfigFileObject;
}
