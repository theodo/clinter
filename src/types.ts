export enum FormatterInfo {
  None = "None",
  Prettier = "Prettier",
}

export interface FormatterInfoObject {
  formatter: FormatterInfo;
}

export enum FrontFrameworkInfo {
  None = "None",
  Vue = "Vue",
  React = "React",
}

export interface FrontFrameworkInfoObject {
  frontFramework: FrontFrameworkInfo;
}

export enum TypescriptInfo {
  None = "None",
  WithTypeChecking = "With type checking",
  NoTypeChecking = "Without type checking",
}

export interface TypescriptInfoObject {
  typescript: TypescriptInfo;
}

export enum EnvInfo {
  Browser = "Browser",
  Node = "Node",
}

export interface EnvInfoObject {
  env: EnvInfo[];
}

export enum TestFrameworkInfo {
  None = "None",
  Jest = "Jest",
}

export interface TestFrameworkInfoObject {
  test: TestFrameworkInfo;
}

export type ProjectInfoObject = FormatterInfoObject &
  FrontFrameworkInfoObject &
  TypescriptInfoObject &
  EnvInfoObject &
  TestFrameworkInfoObject;

export enum ClinterModeInfo {
  Generator = "Generator Mode",
  Upgrade = "Upgrade Mode",
}

export interface ClinterModeInfoObject {
  mode: ClinterModeInfo;
}

export interface MigrationModeInfoObject {
  migration: boolean;
}

export interface DependenciesConfigObject {
  upgradeDependencies: boolean;
}

export interface ClinterSettings {
  generatorConfig: ProjectInfoObject;
  modeConfig: ClinterModeInfoObject;
  migrationModeConfig: MigrationModeInfoObject;
  dependenciesConfig: DependenciesConfigObject;
}
