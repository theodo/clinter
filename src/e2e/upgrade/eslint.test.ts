import { defaultInputConfig, makeTestService } from "e2e/helpers";
import { ClinterModeInfo } from "types";

describe("ESLint Base Configuration Upgrade Mode", () => {
  it.each`
    filename            | initialConfig
    ${".eslintrc"}      | ${"{}"}
    ${".eslintrc.json"} | ${"{}"}
    ${".eslintrc.js"}   | ${"module.exports = {}"}
  `(
    "should return a standard eslint configuration with no extensions appended to an empty config in $filename",
    async ({ filename, initialConfig }: { filename: string; initialConfig: string }) => {
      const testService = makeTestService(filename);

      testService.loadInputConfig({ ...defaultInputConfig, modeConfig: { mode: ClinterModeInfo.Upgrade } });
      testService.loadInitialLinterConfig(initialConfig);
      await testService.runClinter();
      const outputConfig = testService.getOutputConfig();

      expect(outputConfig).toMatchSnapshot(`eslint-simple-config-${filename}`);
    }
  );
});
