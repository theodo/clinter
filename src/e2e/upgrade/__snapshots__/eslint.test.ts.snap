// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Base Configuration Upgrade Mode should return a standard eslint configuration with no extensions appended to an empty config in .eslintrc.js: eslint-simple-config-.eslintrc.js 1`] = `
"module.exports = {
  \\"root\\": true,
  \\"extends\\": [
    \\"eslint:recommended\\"
  ],
  \\"plugins\\": [
    \\"import\\"
  ],
  \\"rules\\": {
    \\"curly\\": [
      \\"error\\",
      \\"all\\"
    ],
    \\"eqeqeq\\": [
      \\"error\\",
      \\"smart\\"
    ],
    \\"complexity\\": [
      \\"error\\",
      8
    ],
    \\"import/no-extraneous-dependencies\\": [
      \\"error\\",
      {
        \\"devDependencies\\": true,
        \\"optionalDependencies\\": false,
        \\"peerDependencies\\": false
      }
    ],
    \\"no-shadow\\": [
      \\"error\\",
      {
        \\"hoist\\": \\"all\\"
      }
    ],
    \\"prefer-const\\": \\"error\\",
    \\"import/order\\": [
      \\"error\\",
      {
        \\"groups\\": [
          [
            \\"external\\",
            \\"builtin\\"
          ],
          \\"internal\\",
          [
            \\"parent\\",
            \\"sibling\\",
            \\"index\\"
          ]
        ]
      }
    ],
    \\"sort-imports\\": [
      \\"error\\",
      {
        \\"ignoreCase\\": true,
        \\"ignoreDeclarationSort\\": true,
        \\"ignoreMemberSort\\": false,
        \\"memberSyntaxSortOrder\\": [
          \\"none\\",
          \\"all\\",
          \\"multiple\\",
          \\"single\\"
        ]
      }
    ],
    \\"padding-line-between-statements\\": [
      \\"error\\",
      {
        \\"blankLine\\": \\"always\\",
        \\"prev\\": \\"*\\",
        \\"next\\": \\"return\\"
      }
    ]
  }
}"
`;

exports[`ESLint Base Configuration Upgrade Mode should return a standard eslint configuration with no extensions appended to an empty config in .eslintrc.json: eslint-simple-config-.eslintrc.json 1`] = `
"{
  \\"root\\": true,
  \\"extends\\": [
    \\"eslint:recommended\\"
  ],
  \\"plugins\\": [
    \\"import\\"
  ],
  \\"rules\\": {
    \\"curly\\": [
      \\"error\\",
      \\"all\\"
    ],
    \\"eqeqeq\\": [
      \\"error\\",
      \\"smart\\"
    ],
    \\"complexity\\": [
      \\"error\\",
      8
    ],
    \\"import/no-extraneous-dependencies\\": [
      \\"error\\",
      {
        \\"devDependencies\\": true,
        \\"optionalDependencies\\": false,
        \\"peerDependencies\\": false
      }
    ],
    \\"no-shadow\\": [
      \\"error\\",
      {
        \\"hoist\\": \\"all\\"
      }
    ],
    \\"prefer-const\\": \\"error\\",
    \\"import/order\\": [
      \\"error\\",
      {
        \\"groups\\": [
          [
            \\"external\\",
            \\"builtin\\"
          ],
          \\"internal\\",
          [
            \\"parent\\",
            \\"sibling\\",
            \\"index\\"
          ]
        ]
      }
    ],
    \\"sort-imports\\": [
      \\"error\\",
      {
        \\"ignoreCase\\": true,
        \\"ignoreDeclarationSort\\": true,
        \\"ignoreMemberSort\\": false,
        \\"memberSyntaxSortOrder\\": [
          \\"none\\",
          \\"all\\",
          \\"multiple\\",
          \\"single\\"
        ]
      }
    ],
    \\"padding-line-between-statements\\": [
      \\"error\\",
      {
        \\"blankLine\\": \\"always\\",
        \\"prev\\": \\"*\\",
        \\"next\\": \\"return\\"
      }
    ]
  }
}"
`;

exports[`ESLint Base Configuration Upgrade Mode should return a standard eslint configuration with no extensions appended to an empty config in .eslintrc: eslint-simple-config-.eslintrc 1`] = `
"{
  \\"root\\": true,
  \\"extends\\": [
    \\"eslint:recommended\\"
  ],
  \\"plugins\\": [
    \\"import\\"
  ],
  \\"rules\\": {
    \\"curly\\": [
      \\"error\\",
      \\"all\\"
    ],
    \\"eqeqeq\\": [
      \\"error\\",
      \\"smart\\"
    ],
    \\"complexity\\": [
      \\"error\\",
      8
    ],
    \\"import/no-extraneous-dependencies\\": [
      \\"error\\",
      {
        \\"devDependencies\\": true,
        \\"optionalDependencies\\": false,
        \\"peerDependencies\\": false
      }
    ],
    \\"no-shadow\\": [
      \\"error\\",
      {
        \\"hoist\\": \\"all\\"
      }
    ],
    \\"prefer-const\\": \\"error\\",
    \\"import/order\\": [
      \\"error\\",
      {
        \\"groups\\": [
          [
            \\"external\\",
            \\"builtin\\"
          ],
          \\"internal\\",
          [
            \\"parent\\",
            \\"sibling\\",
            \\"index\\"
          ]
        ]
      }
    ],
    \\"sort-imports\\": [
      \\"error\\",
      {
        \\"ignoreCase\\": true,
        \\"ignoreDeclarationSort\\": true,
        \\"ignoreMemberSort\\": false,
        \\"memberSyntaxSortOrder\\": [
          \\"none\\",
          \\"all\\",
          \\"multiple\\",
          \\"single\\"
        ]
      }
    ],
    \\"padding-line-between-statements\\": [
      \\"error\\",
      {
        \\"blankLine\\": \\"always\\",
        \\"prev\\": \\"*\\",
        \\"next\\": \\"return\\"
      }
    ]
  }
}"
`;
